### PR TITLE
feat: log reg form info to analytics events

### PR DESCRIPTION
**Pull request by dkoo**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Captures form info to analytics events for forms submitted via prompts. This lets us distinguish between subscribe vs register forms, and tells us whether the reader subscribed to any newsletter lists (and which ones).

### How to test the changes in this Pull Request:

1. Check out this branch, as well as https://github.com/Automattic/newspack-plugin/pull/1953 and https://github.com/Automattic/newspack-newsletters/pull/943
2. Ensure that your site is connected to Google Analytics, and monitor the Realtime events dashboard for incoming events.
3. Publish a prompt containing a registration form.
4. On the front-end, register a new email address and optionally subscribe to one or more newsletter lists.
5. In Google Analytics, look for new events with the event category `Newspack Announcement` and event action `Form Submission`. Confirm that the event label for these events contains not just the prompt info, but also the registration info, in this format:

> Prompt position: Prompt Name (prompt ID) - Segment name | Form ID - signup email - list IDs

6. Publish a prompt containing a subscribe form and repeat steps 4–5.
7. Disable the AMP plugin and repeat steps 4-6.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by dkoo** [comment]
Removing "Needs review" tag because this PR needs to handle non-AMP events as well.

**Comment by dkoo** [comment]
Update: [f34aec3](https://github.com/Automattic/newspack-plugin/pull/1953/commits/f34aec3155cfabd458c931d8db8218addd726eae) polyfills AMP Analytics' handling of dynamic form id and field value placeholders. This is handled on the main plugin side so that it can apply to any incoming analytics events, not just those from this plugin.

File: includes/class-newspack-popups-model.php [hunk 1]
```diff
@@ -968,6 +969,11 @@ function( $acc, $segment_id ) {
 			];
 		}
 
+		// If the form contains registration + list info, append that to the event label.
+		if ( $has_register_form ) {
+			$event_label .= __( ' | ${formId} - ${formFields[npe]} - ${formFields[lists[]]}' );
```
Comment (adekbadek) [inline includes/class-newspack-popups-model.php:974]: It does not seems like a good idea to send PII to GA 🤔

**Review by adekbadek — CHANGES_REQUESTED** [review]

File: includes/class-newspack-popups-model.php (hunk 1 above)
Comment (dkoo) [reply includes/class-newspack-popups-model.php]: Oh, good point—I wasn't thinking of the legal implications of that. 7c0d5fb removes the email from the event info.

**Review by adekbadek — APPROVED** [review]

**Pull request merged**